combat-trainer.lic

```diff
@@ -3859,7 +3858,7 @@ class GameState
     options -= held_types if weapon_skill.eql?('Bow') || (weapon_skill.eql?('Crossbow') && !@using_light_crossbow)
     options -= thrown_types if (weapon_skill.eql?('Bow') && @left_hand_free)
     echo "determine_aiming_skill::options: #{options}" if $debug_mode_ct
-    options.min_by { |skill| [DRSkill.getxp(skill), DRSkill.getrank(skill)] }
+    sort_by_rate_then_rank(options).first
```

Comment:
Original logic already sorted by rate then rank, this change is to make that decision consistent throughout combat-trainer by using `game_state.sort_by_rate_then_rank`. _If_ the decision algoritm should change in the future, there's now one spot to do it.

Reply:
Am I reading this right? It looks like the 'before' version did not `.first`. Should the 'after'?

Reply:
Great question. The `min_by` function implicitly does a `.first` to grab the single item sorted first in the list. `sort_by` returns the whole list but sorted. 

In cases where code did `min_by` originally to pluck a single value then I performed a `.first` in the refactored version.

Reply:
Oh! You're right, I just mixed up the two. Thanks.